### ISSUE TITLE
Add commit message format check to lint pipeline

### ISSUE THREAD

**Issue by hferentschik** [jenkins-x/jx#4284]
Even though the majority of our commits follow the [conventional commits guidelines ](https://www.conventionalcommits.org/en/v1.0.0-beta.3/), every now and then non-conventional commit messages are slipping in. To enforce consistency, I'd like to add commit message format validation to the existing code checks. [Gommit](https://github.com/antham/gommit) could be used for this.[ jx-app-jacoco](https://github.com/jenkins-x-apps/jx-app-jacoco/) already uses it for example.

One issue related to this is our current merge strategy for pull requests. It uses squash instead of rebase. I think there are various issues with that:

* If there are multiple commits, the pull request title is used as the main commit message. Unless the title itself is in conventional commit format one ends up with a non-conventional commit header message
* The auto squashing seems to make developers feel that they don't have to take care about how their commits are crafted since they are getting squashed anyways. However, pushing commits like "fixing Hound errors" will still end up in the commit message making it very messy. Also for each commit squashed we get a duplicated DOC line.

Bottom line, switching to rebase will keep the original commits of the pull request and putting the developer back in charge of crafting consistent and meaningful commits. It will also allow introducing commit message validation. 

@jenkins-x/core ^^ Thoughts?

**Comment by hferentschik** [comment]
Relates to #4282

**Comment by vharsh** [comment]
Gocommit looks like a really great tool for _doing one thing and doing it really well_ and might be better than a tool I knew about that is [pre-commit](https://pre-commit.com/#pre-commit-for-commit-messages) which also has a python dependency.

**Comment by metonymic-smokey** [comment]
@hferentschik is this issue still open?
GSoC aspirant so wanted to pick an issue to start with.

Thanks!

**Comment by metonymic-smokey** [comment]
@jstrachan is this issue still open?
GSoC aspirant here looking to start contributing.

Thanks!
